src/Support/Contracts/ArgBuilderDirective.php

```diff
@@ -7,12 +7,10 @@ interface ArgBuilderDirective extends Directive
     /**
      * Add additional constraints to the builder based on the given argument value.
      *
-     * @template TModel of \Illuminate\Database\Eloquent\Model
```

Comment:
here the generic type makes sense, the method should not change the type of model when returning the eloquent builder